src/Mutator/Util/AbstractAllSubExprNegation.php

```diff
@@ -92,6 +92,15 @@ public function canMutate(Node $node): bool
      */
     abstract protected function getSupportedBinaryOpExprClass(): string;
 
+    protected function isSubConditionMutable(Node\Expr $node): bool
```

Comment:
refactored this base-class to make this logic overridable (no functional changes in this base-class)